dashboard/src/components/OperatorList/OperatorList.v2.test.tsx

```diff
@@ -0,0 +1,262 @@
+});
+
+xit("render the operator list without installed operators", () => {
+  const wrapper = mountWrapper(
+    getStore({ operators: { isOLMInstalled: true, operators: [sampleOperator] } }),
+    <OperatorList {...defaultProps} />,
+  );
+  expect(wrapper.find(OLMNotFound)).not.toExist();
+  expect(wrapper.find(InfoCard)).toExist();
+  // The section "Available operators" should not be empty since the operator is not installed
+  expect(wrapper.find("h3").filterWhere(c => c.text() === "Installed")).not.toExist();
+  expect(
+    wrapper
+      .find(CardGrid)
+      .last()
+      .children(),
+  ).toExist();
+  expect(wrapper).toMatchSnapshot();
+});
+
+describe("filter operators", () => {
+  const sampleOperator2 = {
+    ...sampleOperator,
+    metadata: {
+      name: "bar",
+    },
+    status: {
+      ...sampleOperator.status,
+      channels: [
+        {
+          ...sampleOperator.status.channels[0],
+          currentCSVDesc: {
+            version: "1.0.0",
+            annotations: {
+              categories: "database, other",
+              capabilities: BASIC_INSTALL,
+            },
+          },
+        },
+      ],
+    },
+  } as any;
+
+  it("setting the filter in the state", () => {
+    const wrapper = mountWrapper(
+      getStore({
+        operators: { isOLMInstalled: true, operators: [sampleOperator, sampleOperator2] },
+      }),
+      <OperatorList {...defaultProps} />,
+    );
+    expect(wrapper.find(InfoCard).length).toBe(2);
+    act(() => {
+      (wrapper.find(SearchFilter).prop("onChange") as any)("foo");
+    });
+    wrapper.update();
+    expect(wrapper.find(InfoCard).length).toBe(1);
+  });
+
+  it("setting the filter in the props", () => {
+    const wrapper = mountWrapper(
+      getStore({
+        operators: { isOLMInstalled: true, operators: [sampleOperator, sampleOperator2] },
+      }),
+      <OperatorList {...defaultProps} filter="foo" />,
+    );
+    expect(wrapper.find(InfoCard).length).toBe(1);
+  });
+
+  it("show a message if the filter doesn't match any operator", () => {
+    const wrapper = mountWrapper(
+      getStore({
+        operators: { isOLMInstalled: true, operators: [sampleOperator, sampleOperator2] },
+      }),
+      <OperatorList {...defaultProps} filter="nope" />,
+    );
+    expect(wrapper.find(InfoCard)).not.toExist();
+    expect(wrapper).toIncludeText("No operator matches the current filter");
+  });
+
+  it("filters by category", () => {
+    const wrapper = mountWrapper(
+      getStore({
+        operators: { isOLMInstalled: true, operators: [sampleOperator, sampleOperator2] },
+      }),
+      <OperatorList {...defaultProps} />,
+    );
+    expect(wrapper.find(InfoCard).length).toBe(2);
+
+    // Filter category "security"
+    const input = wrapper.find("input").findWhere(i => i.prop("value") === "security");
+    input.simulate("change", { target: { value: "security" } });
+    expect(wrapper.find(InfoCard).length).toBe(1);
```

Comment:
Should really verify that it's `sampleOperator` right? (I mean, verify that the one item is the correct one).

Reply:
right